### PR TITLE
Adding the ability to customize the HTTP Post payload and use the given handler context

### PR DESCRIPTION
**Pull request by alexjpaz**
I needed to customize the payload to slack and I wasn't able to accomplish it using the `template`  property.

See https://jenkins.moveaws.com/job/prosoft-sensu-base-image/43/console

### CODE REVIEW

**Comment by sensu-plugin** [comment]
@alexjpaz  Can you please update the CHANGELOG as well as bump the patch version and I will merge this. Thanks!

File: bin/handler-slack.rb [hunk 1]
```diff
@@ -114,9 +129,13 @@ def post_data(notice)
...
 
     req = Net::HTTP::Post.new("#{uri.path}?#{uri.query}", 'Content-Type' => 'application/json')
-    text = slack_surround ? slack_surround + notice + slack_surround : notice
 
-    req.body = payload(text).to_json
+    if payload_template.nil?
+      text = slack_surround ? slack_surround + notice + slack_surround : notice
+      req.body = payload(text).to_json
+    else
+      req.body = notice
```
Comment (analytically) [inline bin/handler-slack.rb]: Notice?

Comment (alexjpaz) [inline bin/handler-slack.rb]: I can't change the variable name from "notice" to "body" if that makes more sense.

**Comment by analytically** [comment]
We should explain in the README that `template` is the description template, and payload_template the slack message template. Mind adding this?

**Comment by sstarcher** [comment]
Maybe instead of explaining it changing the variable names to match?
template => description_template
payload_template => message_template

**Comment by analytically** [comment]
Sounds good, but we need to keep --template for backwards compatibility.

**Comment by sensu-plugin** [comment]
Breaking compatibility is something that should only happen on a really important item. I am not against it as long as the version reflects this, which is why i am trying to be even tighter with the CHANGELOG and versioning requirements..

**Comment by sstarcher** [comment]
@analytically agreed, change them both, but also keep --template for backwards compatability.  Just alias it

**Comment by alexjpaz** [comment]
I think something like this should work correct?
```
  def message_template 
    get_setting('template') || get_setting('message_template') 
  end
```

**Comment by analytically** [comment]
Yep!

**Comment by sensu-plugin** [comment]
@alexjpaz  You will need to do a rebase on this to bring in some new bits, nothing that should affect you negatively.

**Comment by sstarcher** [comment]
tested locally since certs are failing on pull requests.

**Pull request merged**